lib/Server.js

```diff
@@ -125,9 +125,12 @@ class Server {
   start() {
     const hostname = this.hostname;
     const port = this.port;
-    this.httpServer = this.server.listen(port, function() {
-      console.log('Server listening at http://%s:%s', hostname, port);
-    });
+    return new Promise(function (resolve) {
+      this.httpServer = this.server.listen(port, function() {
+        console.log('Server listening at http://%s:%s', hostname, port);
+        this._readyPromise.then(resolve);
```

Comment:
> [...] and instead we start everything here in `server.start()`. I think that would be more intuitive.

Yes! I agree.

> I'll try to get that done tonight, then I'll pull in your changes :smile: 

Or perhaps the other way around, to avoid conflicts? Your call. :)